### PR TITLE
statement.d: for => foreach

### PR DESCRIPTION
**Pull request by WalterBright**
refactor only

### CODE REVIEW

**Comment by yebblies** [comment]
What do you think about adding a simple opApply(index, T) to Array, so we can use it with foreach?

File: src/statement.d [hunk 1]
```diff
@@ -3787,7 +3787,7 @@ public:
             }
             sw.cases.push(this);
             // Resolve any goto case's with no exp to this case statement
-            for (size_t i = 0; i < sw.gotoCases.dim;)
+            foreach (i; 0 .. sw.gotoCases.dim)
```
Comment (CyberShadow) [inline src/statement.d]: This doesn't look right :)

Comment (WalterBright) [inline src/statement.d]: yeah, that looks suspicious

Comment (yebblies) [inline src/statement.d]: I think `foreach (ref i; 0 .. sw.gotoCases.dim)` will do what you want, but I'm not sure that's in 2.067.

**Comment by yebblies** [comment]
Add this to `root.Array`:
```d
    extern (D) T[] opSlice()
    {
        return data[0..dim];
    }
```
Then it becomes `foreach (i, s; cs.statements)`

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**